test/WordsAndDocsFileParserTest.cpp

```diff
@@ -10,59 +10,109 @@
 #include "./WordsAndDocsFileLineCreator.h"
 #include "parser/WordsAndDocsFileParser.h"
 
+// All lambdas and type aliases used in this file contained here
+namespace {
+
+/// Type aliases
+
+// Word, isEntity, contextId, score
+using WordLine = std::tuple<std::string, bool, size_t, size_t>;
+using WordLineVec = std::vector<WordLine>;
+
+// docId, docContent
+using DocLine = std::tuple<size_t, std::string>;
+using DocLineVec = std::vector<DocLine>;
+
+using StringVec = std::vector<std::string>;
+
+/// Lambdas
+
+auto getLocaleManager = []() { return LocaleManager("en", "US", false); };
+
+auto wordsFileLineToWordLine = [](WordsFileLine wordsFileLine) -> WordLine {
+  return std::make_tuple(wordsFileLine.word_, wordsFileLine.isEntity_,
+                         static_cast<size_t>(wordsFileLine.contextId_.get()),
+                         static_cast<size_t>(wordsFileLine.score_));
+};
+
+// Lambda that takes in a path to wordsFile to initialize the Parser and an
+// expectedResult that is compared against the parsers outputs.
+auto testWordsFileParser = [](std::string wordsFilePath,
+                              WordLineVec expectedResult) {
+  WordsFileParser p(wordsFilePath, getLocaleManager());
+  size_t i = 0;
+  for (auto wordsFileLine : p) {
+    ASSERT_TRUE(i < expectedResult.size());
+    WordLine testLine = wordsFileLineToWordLine(wordsFileLine);
+
+    // Not testing the whole tuples against each other to have a cleaner
+    // indication what exactly caused the assertion to fail
+    ASSERT_EQ(std::get<0>(testLine), std::get<0>(expectedResult.at(i)));
+    ASSERT_EQ(std::get<1>(testLine), std::get<1>(expectedResult.at(i)));
+    ASSERT_EQ(std::get<2>(testLine), std::get<2>(expectedResult.at(i)));
+    ASSERT_EQ(std::get<3>(testLine), std::get<3>(expectedResult.at(i)));
```

Comment:
That is much better with the helper functions (There are even cleaner ways with better error messages in GoogleTest, but this refactoring is nice because now all improviements can be applied locally!